### PR TITLE
Only produce qrsearchers config with local dispatcher present.

### PR DESCRIPTION
**Pull request by baldersheim**
@hmusum, @bratseth or @jobergum or any other inclined to get dirty

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/search/ContainerSearch.java [hunk 1]
```diff
@@ -178,9 +182,11 @@ public void getConfig(QrSearchersConfig.Builder builder) {
     		if (sys instanceof IndexedSearchCluster) {
     			scB.rowbits(sys.getRowBits());
                 for (Dispatch tld: ((IndexedSearchCluster)sys).getTLDs()) {
-                	scB.dispatcher(new QrSearchersConfig.Searchcluster.Dispatcher.Builder().
-                			host(tld.getHostname()).
-                			port(tld.getDispatchPort()));
+                    if (hostName.equals(tld.getHostname())) {
+                        scB.dispatcher(new QrSearchersConfig.Searchcluster.Dispatcher.Builder().
+                                host(tld.getHostname()).
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/vespa/model/container/search/ContainerSearch.java]: Now the hostname could be replaced with "localhost". Not sure if that is desirable or not.

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerCluster.java [hunk 2]
```diff
@@ -138,7 +135,7 @@
 
     /**
      * URI prefix used for internal, usually programmatic, APIs. URIs using this
-     * prefix should never considered available for direct use by customers, and
+Qr     * prefix should never considered available for direct use by customers, and
```
Comment (hmusum) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerCluster.java]: Remove unintentionally added characters

**Review by hmusum — CHANGES_REQUESTED** [review]
Except the invalid syntax due to stray characters, LGTM

**Comment by baldersheim** [comment]
Given that this is the output from configproxy
{code}
[0] honda:~/git/systemtests/tests/search/reportcoverage$ vespa-configproxy-cmd |grep ClusterSea | grep qr-searchers
container.qr-searchers,combinedcontainer/searchchains/chain/search/component/com.yahoo.prelude.cluster.ClusterSearcher,3,22f0ecd3defd7c9b0c37760b2311d412
container.qr-searchers,doc-api/searchchain/chain/search/component/com.yahoo.prelude.cluster.ClusterSearcher,3,22f0ecd3defd7c9b0c37760b2311d412
{code}

I guess this approach is not possible. There is nothing in the config id that will enable it to reach the instance. In that case this will never work as it has no idea about hostname.
@gjoranv Or is there any magic I am not aware of.
I will build and run a system test first.

**Review by bratseth — APPROVED** [review]

**Comment by baldersheim** [comment]
Closing as it seems this was a dead end

**Comment by gjoranv** [comment]
@baldersheim The components are children of the ContainerCluster, not the Container. Why did you have to move the config generation to Container?

**Comment by baldersheim** [comment]
Because I wanted to limit the set of dispatch nodes to only contain the ones on the same host. So the hostname was needed. But since that was not possible, I did it in the ClusterSearcher instead.

**Pull request closed without merging**